amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/rm/rest/api/holds/HoldsEntityResource.java

```diff
@@ -0,0 +1,184 @@
+import org.alfresco.repo.transaction.RetryingTransactionHelper.RetryingTransactionCallback;
+import org.alfresco.rest.framework.Operation;
+import org.alfresco.rest.framework.WebApiDescription;
+import org.alfresco.rest.framework.WebApiParam;
+import org.alfresco.rest.framework.resource.EntityResource;
+import org.alfresco.rest.framework.resource.actions.interfaces.EntityResourceAction;
+import org.alfresco.rest.framework.resource.parameters.Parameters;
+import org.alfresco.rest.framework.webscripts.WithResponse;
+import org.alfresco.rm.rest.api.impl.ApiNodesModelFactory;
+import org.alfresco.rm.rest.api.impl.FilePlanComponentsApiUtils;
+import org.alfresco.rm.rest.api.model.HoldDeletionReason;
+import org.alfresco.rm.rest.api.model.HoldModel;
+import org.alfresco.service.cmr.model.FileFolderService;
+import org.alfresco.service.cmr.model.FileInfo;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.transaction.TransactionService;
+import org.springframework.beans.factory.InitializingBean;
+
+/**
+ * Hold entity resource
+ *
+ * @author Damian Ujma
+ */
+@EntityResource(name = "holds", title = "Holds")
+public class HoldsEntityResource implements
+    EntityResourceAction.ReadById<HoldModel>,
+    EntityResourceAction.Update<HoldModel>,
+    EntityResourceAction.Delete,
+    InitializingBean
+{
+    private FilePlanComponentsApiUtils apiUtils;
+    private FileFolderService fileFolderService;
+    private ApiNodesModelFactory nodesModelFactory;
+    private HoldService holdService;
+    private TransactionService transactionService;
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        mandatory("nodesModelFactory", nodesModelFactory);
+        mandatory("apiUtils", apiUtils);
+        mandatory("fileFolderService", fileFolderService);
+        mandatory("holdService", holdService);
+        mandatory("transactionService", transactionService);
+    }
+
+    @Override
+    @WebApiDescription(title = "Get hold information", description = "Get information for a hold with id 'holdId'")
+    @WebApiParam(name = "holdId", title = "The hold id")
+    public HoldModel readById(String holdId, Parameters parameters)
+    {
+        checkNotBlank("holdId", holdId);
+        mandatory("parameters", parameters);
+
+        NodeRef hold = apiUtils.lookupAndValidateNodeType(holdId, RecordsManagementModel.TYPE_HOLD);
+        FileInfo info = fileFolderService.getFileInfo(hold);
+        return nodesModelFactory.createHoldModel(info);
+    }
+
+    @Override
+    @WebApiDescription(title = "Update a hold", description = "Updates a hold with id 'holdId'")
+    public HoldModel update(String holdId, HoldModel holdModel, Parameters parameters)
+    {
+        checkNotBlank("holdId", holdId);
+        mandatory("holdModel", holdModel);
+        mandatory("holdModel.name", holdModel.getName());
+        mandatory("holdModel.description", holdModel.getDescription());
```

Comment:
Again I didn't think this was mandatory.

Reply:
Parameter check for the description removed.